src/wp-includes/class-wp-query.php

```diff
@@ -3054,6 +3059,55 @@ public function get_posts() {
 		 */
 		$this->posts = apply_filters_ref_array( 'posts_pre_query', array( null, &$this ) );
 
+		$cache = false;
+		if ( $q['post_query_cache'] ) {
+			$cache_args = $q;
+
+			unset(
+				$cache_args['suppress_filters'],
+				$cache_args['cache_results'],
+				$cache_args['fields'],
+				$cache_args['update_post_meta_cache'],
+				$cache_args['update_post_term_cache'],
+				$cache_args['lazy_load_term_meta']
+			);
+			$key          = md5( serialize( $cache_args ) );
+			$last_changed = wp_cache_get_last_changed( 'posts' );
+			$cache_key    = "wp_query:$key:$last_changed";
+			$cache        = wp_cache_get( $cache_key, 'posts' );
+		}
+		if ( null === $this->posts && $cache ) {
+			if ( 'ids' === $q['fields'] ) {
+				/** @var int[] */
+				$this->posts = array_map( 'intval', $cache['posts'] );
+			} else {
+				_prime_post_caches( $cache['posts'], $q['update_post_term_cache'], $q['update_post_meta_cache'] );
+				/** @var WP_Post[] */
+				$this->posts = array_map( 'get_post', $cache['posts'] );
+			}
+			$this->post_count    = count( $this->posts );
+			$this->found_posts   = $cache['found_posts'];
+			$this->max_num_pages = $cache['max_num_pages'];
+
+			if ( 'id=>parent' === $q['fields'] && $cache ) {
+				/** @var int[] */
+				$r = array();
+				foreach ( $this->posts as $key => $post ) {
+					$obj              = new stdClass();
+					$obj->ID          = (int) $post->ID;
+					$obj->post_parent = (int) $post->post_parent;
+
+					$this->posts[ $key ] = $obj;
+
+					$r[ (int) $post->ID ] = (int) $post->post_parent;
```

Comment:
Here can we use `$obj->ID` and `$obj->post_parent` as it was already get `int` value?